src/flow/boykov_kolmogorov.jl

```diff
@@ -60,39 +63,39 @@ function boykov_kolmogorov_impl{T<:Number}(
     return flow, flow_matrix, TREE
 end
 
-function find_path!{T<:Number}(
-    residual_graph::DiGraph,               # the input graph
-    source::Int,                           # the source vertex
-    target::Int,                           # the target vertex
-    flow_matrix::AbstractArray{T,2},       # the current flow matrix
-    capacity_matrix::AbstractArray{T,2},   # edge flow capacities
-    PARENT::Vector{Int},                   # parent table
-    TREE::Vector{Int},                     # tree table
-    A::Vector{Int}                         # active set
+@traitfn function find_path!{G<:AbstractGraph; IsDirected{G}}(
+    residual_graph::G,            # the input graph
+    source::Integer,                       # the source vertex
+    target::Integer,                       # the target vertex
+    flow_matrix::AbstractMatrix,       # the current flow matrix
+    capacity_matrix::AbstractMatrix,   # edge flow capacities
+    PARENT::Vector,                     # parent table
+    TREE::Vector,                       # tree table
+    A::Vector                           # active set
```

Comment:
I am ok leaving them as untyped and letting MethodNotFound errors happen naturally if we find a bug report, then we can handle it then.